pkg/attribute/bag_test.go

```diff
@@ -567,6 +568,119 @@ func TestMessageDictEdge(t *testing.T) {
 	}
 }
 
+func TestDoubleStrings(t *testing.T) {
+	globalWordList := []string{"HELLO"}
+	globalDict := map[string]int32{globalWordList[0]: 0}
+	messageWordList := []string{"HELLO", "GOOD", "BAD"}
+
+	attrs := mixerpb.Attributes{
+		Words:   messageWordList,
+		Strings: map[int32]int32{0: -3, -1: -2},
+	}
+
+	for i := 0; i < 2; i++ {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			var b Bag
+
+			if i == 0 {
+				b = NewProtoBag(&attrs, globalDict, globalWordList)
+			} else {
+				var err error
+				b, err = GetBagFromProto(&attrs, globalWordList)
+				if err != nil {
+					t.Error("Expecting success, got %v", err)
+				}
+			}
+
+			v, ok := b.Get("HELLO")
+			if !ok {
+				t.Error("Expecting true, got false")
+			}
+
+			s, ok := v.(string)
+			if !ok {
+				t.Errorf("Expecting to get a string, got %v", v)
+			}
+
+			if s != "GOOD" {
+				t.Errorf("Expecting GOOD got %s", s)
+			}
+		})
+	}
+}
+
+func TestReferenceTracking(t *testing.T) {
+	globalWordList := []string{"G0", "G1", "G2"}
+	globalDict := map[string]int32{globalWordList[0]: 0, globalWordList[1]: 1, globalWordList[2]: 2}
+	messageWordList := []string{"N1", "N2", "N3"}
+
+	attrs := mixerpb.Attributes{
+		Words: messageWordList,
+		Strings: map[int32]int32{
+			0:  0,
```

Comment:
can we add comments about the intended affect of these maps? I find it hard to follow.